server/src/controllers/auth.controller.js

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import '@babel/polyfill';
+import models from '../database/models';
+import { encrpytPassword, generateToken, comparePassword } from '../helper/encryption';
+import getSignupUserData from '../utils/user.utils';
+
+
+const { Users } = models;
+
+const signupUser = async (req, res) => {
+  const userData = getSignupUserData(req.body);
+
+  const existingUser = await Users.findOne({
+    where: {
+      email: userData.email
+    }
+  });
+
+  if (existingUser) {
+    return res.status(409).json({ error: 'User with that email already exists' });
+  }
+
+  const hashPassword = await encrpytPassword(userData.password);
+
+  const newUser = await Users.create({ ...userData, password: hashPassword });
+
+  const payload = {
+    newUser
+  };
+
+  const token = generateToken(payload);
+
+  newUser.password = undefined;
```

Comment:
Would this still return the `password` field in the user object but as `undefined`?

Reply:
it won't return it as undefined, it will not appear in the object
![Uploading Screenshot 2020-04-29 at 19.33.30.png…]()